src/components/NodeEditor/NodeEditor.tsx

```diff
@@ -228,12 +319,100 @@ export default class NodeEditor extends React.PureComponent<NodeEditorProps, Nod
         );
     }
 
+    private getInitialState(): NodeEditorState {
```

Comment:
moving state composition to its own method, hoping this improves readability for others as well